novawallet/Modules/DelegatedAccounts/Multisig/MultisigTxDetails/Factory/MultisigTxDetailsViewModelFactory.swift

```diff
@@ -0,0 +1,229 @@
+            imageViewModel: nil
+        )
+        let callHashField = MultisigTxDetailsViewModel.SectionField(
+            title: R.string.localizable.commonCallHash(preferredLanguages: locale.rLanguages),
+            value: callHashModel
+        )
+
+        var callDataField: MultisigTxDetailsViewModel.SectionField<StackCellViewModel>?
+
+        if let callData = multisigTxDetails.callData {
+            let callDataModel = StackCellViewModel(
+                details: callData.toHexWithPrefix(),
+                imageViewModel: nil
+            )
+
+            callDataField = MultisigTxDetailsViewModel.SectionField(
+                title: R.string.localizable.commonCallData(preferredLanguages: locale.rLanguages),
+                value: callDataModel
+            )
+        }
+
+        let sectionModel = MultisigTxDetailsViewModel.CallData(
+            callHash: callHashField,
+            callData: callDataField
+        )
+
+        return .callData(sectionModel)
+    }
+
+    func createCallJSONSection(
+        prettifiedJsonString: String?,
+        locale: Locale
+    ) -> MultisigTxDetailsViewModel.Section? {
+        guard let prettifiedJsonString else { return nil }
+
+        let title = R.string.localizable.operationDetailsCheckHint(
+            preferredLanguages: locale.rLanguages
+        )
+        let field = MultisigTxDetailsViewModel.SectionField(
+            title: title,
+            value: prettifiedJsonString
+        )
+
+        return .callJson(field)
+    }
+
+    func createDepositorViewModel(
+        using account: MultisigTxDetails.Account,
+        chain: ChainModel
+    ) throws -> DisplayAddressViewModel {
+        switch account {
+        case let .local(localAccount):
+            let imageViewModel = iconViewModelFactory.createDrawableIconViewModel(
+                from: localAccount.walletIdenticonData
+            )
+
+            return DisplayAddressViewModel(
+                address: try localAccount.chainAccount.accountId.toAddress(using: chain.chainFormat),
+                name: localAccount.chainAccount.name,
+                imageViewModel: imageViewModel
+            )
+        case let .remote(accountId):
+            let imageViewModel = iconViewModelFactory.createIdentifiableDrawableIconViewModel(
+                from: accountId,
+                chainFormat: chain.chainFormat
+            )
+
+            return DisplayAddressViewModel(
+                address: try accountId.toAddress(using: chain.chainFormat),
+                name: nil,
+                imageViewModel: imageViewModel
+            )
+        }
+    }
+
+    func createDepositViewModel(
+        deposit: BigUInt,
+        depositAsset: ChainAsset,
+        assetPrice: PriceData?,
+        locale: Locale
+    ) -> BalanceViewModelProtocol {
+        let assetInfo = depositAsset.asset.displayInfo
+
+        let depositDecimal = Decimal.fromSubstrateAmount(
```

Comment:
`deposit.decimal(assetInfo: assetInfo)`. Could you, please, also refactor other places in the file if there is any